src/cpp/subscriber/QosSerializer.cpp

```diff
@@ -449,72 +449,76 @@ void serialize<fastdds::dds::ParameterPropertyList_t> (
 }
 
 database::Qos reader_proxy_data_to_backend_qos(
-        const fastrtps::rtps::ReaderProxyData& reader_data)
+        const fastdds::rtps::SubscriptionBuiltinTopicData& reader_data)
 {
     database::Qos reader;
 
-    serialize(reader_data.m_qos.m_durability, durability_tag, reader);
-    serialize(reader_data.m_qos.m_deadline, deadline_tag, reader);
-    serialize(reader_data.m_qos.m_latencyBudget, latency_budget_tag, reader);
-    serialize(reader_data.m_qos.m_liveliness, liveliness_tag, reader);
-    serialize(reader_data.m_qos.m_reliability, reliability_tag, reader);
-    serialize(reader_data.m_qos.m_ownership, ownership_tag, reader);
-    serialize(reader_data.m_qos.m_destinationOrder, destination_order_tag, reader);
-    serialize(reader_data.m_qos.m_userData, user_data_tag, reader);
-    serialize(reader_data.m_qos.m_timeBasedFilter, time_based_filter_tag, reader);
-    serialize(reader_data.m_qos.m_presentation, presentation_tag, reader);
-    serialize(reader_data.m_qos.m_partition, partition_tag, reader);
-    serialize(reader_data.m_qos.m_topicData, topic_data_tag, reader);
-    serialize(reader_data.m_qos.m_groupData, group_data_tag, reader);
-    serialize(reader_data.m_qos.m_durabilityService, durability_service_tag, reader);
-    serialize(reader_data.m_qos.m_lifespan, lifespan_tag, reader);
-    serialize(reader_data.m_qos.representation, representation_tag, reader);
-    serialize(reader_data.m_qos.type_consistency, type_consistency_tag, reader);
-    serialize(reader_data.m_qos.m_disablePositiveACKs, disable_positive_acks_tag, reader);
-    serialize(reader_data.m_qos.data_sharing, data_sharing_tag, reader);
+    // TODO: Implement serialization when available in Fast DDS
+    // serialize(reader_data.durabilityService, durability_service_tag, reader);
```

Comment:
What should we do in this case scenario?

Reply:
DurabilityServiceQosPolicy is not implemented yet in Fast DDS. I think we only need to uncomment line 457 and it's done.